internal/charmstore/migrations_test.go

```diff
@@ -825,49 +333,55 @@ func (s *migrationsSuite) checkBaseEntitiesCount(c *gc.C, expectCount int) {
 	c.Assert(count, gc.Equals, expectCount)
 }
 
-func (s *migrationsSuite) insertEntity(c *gc.C, id *charm.Reference, name string, size int64) {
-	entity := &mongodoc.Entity{
-		URL:     id,
-		BaseURL: baseURL(id),
-		Name:    name,
-		Size:    size,
+// denormalizeEntity returns a copy of e0 with all denormalized fields
+// filled out if they are zero.
+func denormalizeEntity(e0 *mongodoc.Entity) *mongodoc.Entity {
+	e := *e0
+	if e.BaseURL == nil {
+		e.BaseURL = baseURL(e.URL)
 	}
-	err := s.db.Entities().Insert(entity)
-	c.Assert(err, gc.IsNil)
-
-	// Remove the denormalized fields if required.
-	if name != "" {
-		return
+	if e.Name == "" {
```

Comment:
It is documented: "all denormalized fields filled out if they are zero".
The empty string is "zero" in Go parlance.